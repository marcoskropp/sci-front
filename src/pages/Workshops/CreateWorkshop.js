import React, { useState, forwardRef } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Grid,
  Paper,
  Button,
  Container,
  TextField,
  Typography
} from '@material-ui/core';

import withBars from '../../hocs/withBars';
import withAuthChecking from '../../hocs/withAuthChecking';

import { store } from '../../services/Workshops.services';

function CreateWorkshop(props) {
  const [workshop, setWorkshop] = useState({
    title: '',
    description: '',
    place: '',
    startDate: '',
    endDate: ''
  });

  const link = forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

  const handleSubmit = async event => {
    event.preventDefault();

    await store(workshop);

    props.history.push('/');
  };

  const handleChange = event => {
    const { name, value } = event.target;

    setWorkshop({ ...workshop, [name]: value });
  };

  const renderForm = () => {
    const { title, description, place, startDate, endDate } = workshop;

    return (
      <form onSubmit={handleSubmit}>
        <Grid container spacing={3} justify="center">
          <Grid item xs={12}>
            <TextField
              fullWidth
              label="Título"
              name="title"
              value={title}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Descrição"
              name="description"
              value={description}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Local"
              name="place"
              onChange={handleChange}
              value={place}
              variant="outlined"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Data de início"
              name="startDate"
              onChange={handleChange}
              value={startDate}
              variant="outlined"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Data de término"
              name="endDate"
              onChange={handleChange}
              value={endDate}
              variant="outlined"
            />
          </Grid>
          <Grid item xs={12}>
            <Button type="submit" variant="contained" color="primary">
              Cadastrar
            </Button>
            &nbsp;
            <Button
              variant="contained"
              color="secondary"
              to="/"
              component={link}
            >
              Voltar
            </Button>
          </Grid>
        </Grid>
      </form>
    );
  };

  return (
    <Container>
      <Paper style={{ padding: '30px' }}>
        <Typography variant="h4" component="h2" gutterBottom>
          REGISTRAR
        </Typography>
        {renderForm()}
      </Paper>
    </Container>
  );
}

CreateWorkshop.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  })
};


export default withBars(withAuthChecking(CreateWorkshop));
