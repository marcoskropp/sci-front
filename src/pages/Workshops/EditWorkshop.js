import React, { useState, useEffect, forwardRef } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Grid,
  Paper,
  Button,
  Container,
  TextField,
  Typography
} from '@material-ui/core';

import { update } from '../../services/Workshops.services';

export default function EditWorkshop(props) {
  const workshopId = props.match.params.id;

  const [workshop, setWorkshop] = useState({
    id: 0,
    title: '',
    description: '',
    place: '',
    startDate: '',
    endDate: ''
  });

  useEffect(() => {
    const getWorkshop = async () => {
      const myWorkshop = await update(workshopId);
      setWorkshop({ ...myWorkshop });
    };
    getWorkshop();
  }, [workshopId]);

  const link = forwardRef((props, ref) => (
    <Link innerRef={ref} {...props} />
  ));

  const handleChange = event => {
    const { name, value } = event.target;

    setWorkshop({ ...workshop, [name]: value });
  };

  const renderForm = () => {
    const { title, description, place, startDate, endDate } = workshop;

    return (
      <Grid container spacing={3} justify="center">
        <Grid item xs={12}>
          <TextField
            fullWidth
            label="Título"
            name="title"
            value={title}
            variant="outlined"
            onChange={handleChange}
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            label="Descrição"
            name="description"
            value={description}
            variant="outlined"
            onChange={handleChange}
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            label="Local"
            name="place"
            onChange={handleChange}
            value={place}
            variant="outlined"
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            label="Data de início"
            name="startDate"
            onChange={handleChange}
            value={startDate}
            variant="outlined"
            type="date"
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            label="Data de término"
            name="endDate"
            onChange={handleChange}
            value={endDate}
            variant="outlined"
            type="date"
          />
        </Grid>
        <Grid item xs={12}>
          <Button
            variant="contained"
            color="secondary"
            to="/"
            component={link}
          >
                        Voltar
          </Button>
        </Grid>
      </Grid>
    );
  };

  return (
    <Container>
      <Paper style={{ padding: '30px' }}>
        <Typography variant="h4" component="h2" gutterBottom>
                    EDITAR
        </Typography>
        {renderForm()}
      </Paper>
    </Container>
  );
}

EditWorkshop.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number
    })
  })
};
