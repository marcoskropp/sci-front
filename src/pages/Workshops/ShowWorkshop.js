import React, { useState, useEffect, forwardRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Grid,
  Paper,
  Button,
  Container,
  TextField,
  Typography,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from '@material-ui/core';

import withBars from '../../hocs/withBars';
import withAuthChecking from '../../hocs/withAuthChecking';

import { show } from '../../services/Workshops.services';

const useStyles = makeStyles({
  root: {
    width: '100%',
    overflowX: 'auto',
    marginTop: 30
  },
  table: {
    minWidth: 650
  }
});

function ShowWorkshop(props) {
  const classes = useStyles();

  const [workshop, setWorkshop] = useState({});

  useEffect(() => {
    const getWorkshop = async () => {
      const myWorkshop = await show(props.match.params.id);
      setWorkshop({ ...myWorkshop });
    };
    getWorkshop();
  }, [props.match.params.id]);

  const link = forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

  const renderForm = () => {
    const { title, description, place, startDate, endDate } = workshop;

    return (
      <Grid container spacing={3} justify="center">
        <Grid item xs={12}>
          <TextField
            fullWidth
            name="title"
            value={title}
            variant="outlined"
            disabled
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            name="description"
            value={description}
            variant="outlined"
            disabled
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            name="place"
            value={place}
            variant="outlined"
            disabled
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            name="startDate"
            value={startDate}
            variant="outlined"
            type="text"
            disabled
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            fullWidth
            name="endDate"
            value={endDate}
            variant="outlined"
            type="text"
            disabled
          />
        </Grid>
      </Grid>
    );
  };

  const renderTable = () => (
    <Table className={classes.table} aria-label="simple table">
      <TableHead>
        <TableRow>
          <TableCell>Incritos</TableCell>
          <TableCell align="right">Email</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {workshop.users
            && workshop.users.map(({ id, name, email }) => (
              <TableRow key={id}>
                <TableCell component="th" scope="row">
                  {name}
                </TableCell>
                <TableCell align="right">{email}</TableCell>
              </TableRow>
            ))}
      </TableBody>
    </Table>
  );

  return (
    <Container>
      <Paper style={{ padding: '30px' }}>
        <Typography variant="h4" component="h2" gutterBottom>
          VISUALIZAR
        </Typography>
        {renderForm()}
        {renderTable()}
        <Grid item xs={12}>
          <Button variant="contained" color="secondary" to="/" component={link}>
            Voltar
          </Button>
        </Grid>
      </Paper>
    </Container>
  );
}

ShowWorkshop.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number
    })
  })
};


export default withBars(withAuthChecking(ShowWorkshop));
