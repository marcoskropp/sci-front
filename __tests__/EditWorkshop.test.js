import React from 'react';
import { shallow } from 'enzyme';

import { EditWorkshop } from '../src/pages/Workshops/EditWorkshop';

const props = {
  match: {
    params: {
      id: 1
    }
  }
};


describe('Testing EditWorkshop Page', () => {
  it('Snapshot renders', () => {
    const wrapper = shallow(<EditWorkshop {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
