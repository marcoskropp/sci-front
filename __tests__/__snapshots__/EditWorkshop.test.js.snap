// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditWorkshop Page Snapshot renders 1`] = `ShallowWrapper {}`;
